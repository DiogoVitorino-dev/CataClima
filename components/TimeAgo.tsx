import { formatDistanceToNow,Locale, parseISO } from "date-fns"
import { OpenTextDefault } from "./StyledText"

const localeTranslated = {
    code:'pt',
    formatDistance:(message:string,num:number) => { 
                       
        message = message.replace(/about/i,'Há')        
        message = message.replace(/lessThanXMinutes/i,'Há poucos segundos')

        message = message.replace(/Hours/i,'Hora')
        message = message.replace(/Minutes/i,'minuto')
        message = message.replace(/Month/i,'mês')
        message = message.replace(/Year/i,'ano')
        
        if (num > 1){                             
            message = message.replace('mês','mese')
            message += 's'
        }
        return message.replace(/x/i,` ${num} `) + ' atrás'
    }                   
} as Locale

export default function TimeAgo({time}:{time:string}) {
    time = formatDistanceToNow(parseISO(time),{locale:localeTranslated}) 
    

    return (
        <OpenTextDefault style={{
            fontSize:14,
            color:'#fff',
            opacity:0.8
        }}>
            {time}  
        </OpenTextDefault>
    )        
}