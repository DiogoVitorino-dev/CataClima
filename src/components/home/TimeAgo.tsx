import WeatherText from "./WeatherText";

import Colors from "@/constants/Colors";
import { DateTimeService } from "@/services/DateTimeService";

interface IProps {
  date: string;
}

export default function TimeAgo({ date }: IProps) {
  const color = Colors["home"];
  return (
    <WeatherText
      style={{
        fontSize: 14,
      }}
      color={color.subtitle}
    >
      {DateTimeService.formatTimeDistance(date)}
    </WeatherText>
  );
}
