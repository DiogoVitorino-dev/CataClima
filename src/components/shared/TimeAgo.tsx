import { formatDistanceToNow,Locale, parseISO } from 'date-fns';
import { OpenText } from './StyledText';

const localeTranslated = {
	code:'pt',
	formatDistance:(message:string,num:number) => { 
                       
		message = message.replace(/about/i,'Há');        
		message = message.replace(/lessThanXMinutes/i,'Há poucos segundos');

		message = message.replace(/Hours/i,'Hora');
		message = message.replace(/Minutes/i,'minuto');
		message = message.replace(/Days/i,'dia');
		message = message.replace(/Weeks/i,'semana');
		message = message.replace(/Month/i,'mês');
		message = message.replace(/Year/i,'ano');
        
		if (num > 1){                             
			message = message.replace('mês','mese');
			message += 's';
		}
		return message.replace(/x/i,` ${num} `) + ' atrás';
	}                   
} as Locale;

export function TimeAgo({dateIsoFormat}:{dateIsoFormat:string}) {

	const formatDate = () => {
		try {
			const dateFormat = parseISO(dateIsoFormat);
			return formatDistanceToNow(dateFormat,{locale:localeTranslated});
		} catch (error:any) {
			(error.message);            
		}

		return '';
	};

	return (
		<OpenText style={{
			fontSize:14,
			color:'#fff',
			opacity:0.8
		}}>
			{formatDate()}  
		</OpenText>
	);        
}