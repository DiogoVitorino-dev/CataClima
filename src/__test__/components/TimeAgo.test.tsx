import {describe,it, expect, afterEach} from '@jest/globals';
import { cleanup } from '@testing-library/react-native';
import renderer from 'react-test-renderer'
import TimeAgo from '../../components/TimeAgo';

describe('TimeAgo component', () => {    
    afterEach(cleanup)

    it('should renders correctly', () => {
        const date = new Date().toISOString()
        const tree = renderer.create(
        <TimeAgo dateIsoFormat={date} />).toJSON();
        expect(tree).toMatchSnapshot()    
    })
})
