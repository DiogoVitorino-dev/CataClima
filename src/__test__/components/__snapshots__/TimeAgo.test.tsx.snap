// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeAgo component should renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#fff",
        "fontSize": 14,
        "opacity": 0.8,
      },
      {
        "fontFamily": "OpenSans",
      },
    ]
  }
>
  Há poucos segundos atrás
</Text>
`;
